SplunkRumWorkspace/SplunkRum/SplunkRum/SessionBasedSampler.swift

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2021 Splunk Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import Foundation
+import OpenTelemetrySdk
+import OpenTelemetryApi
+
+class SessionBasedSampler {
+    static var probability: Double = 1.0
+    static var sessionCount: Int = 0
+    static var timer: Timer?
+
+    init(ratio: Double) {
+        SessionBasedSampler.probability = ratio
+    }
+
+    /**Check if session will be sampled or not.**/
+    @discardableResult public class func sessionShouldSample() -> Bool {
+        let samplingPercentage = SessionBasedSampler.probability
+        let step = Double(1/samplingPercentage)
+        let roundedStepValue = round(step * 10) / 10.0
+
+        var result = false
+        switch SessionBasedSampler.probability {
+        case 0.0:
+            result = false
+        case 1.0:
+            result = true
+        default:
+            if floor(Double(SessionBasedSampler.sessionCount).truncatingRemainder(dividingBy: roundedStepValue)) == 0 {
```

Comment:
Why isn't this just rolling an RNG?  The way I read this the 1st session (sessionCount == 0) will always be sampled. 
Pseudocode:
```
result = (((float) arc4random()) / UNIT32_MAX) <= probability;
```